src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -214,21 +217,27 @@ public override void Start()
         /// </summary>
         private void StartAddressManager(NodeConnectionParameters connectionParameters)
         {
-            this.addressManager = new PeerAddressManager(this.asyncLoopFactory, this.dataFolder.AddressManagerFilePath);
-            connectionParameters.TemplateBehaviors.Add(new PeerAddressManagerBehaviour(this.addressManager));
+            this.addressManager = new PeerAddressManager(this.dataFolder);
+            var addressManagerBehaviour = new PeerAddressManagerBehaviour(this.addressManager);
```

Comment:
`PeerAddressManager` and `PeerAddressManagerBehaviour` can both be injected in the feature constructor.

Reply:
Just to be clear, are we then DI-ing the PeerAddressManager and Behaviour?

I.e. we then wont be new-ing up the instances in StartAddressManager?

Reply:
Exactly, we leave that to the DI, that's the best practice at least to let the DI manage instance creation. its also beneficial for testing.